File: SrdDatabase/Models/Payments/Payment.cs
using System;

namespace SrdDatabase.Models.Payments
{
    public class Payment
    {
        public int Id { get; }

        public int Amount { get; }

        public DateTime Date { get; }

        public int? ReceiptNumber { get; }

        public int CongregationId { get; }

        public string Congregation { get; }

        public int ParishId { get; }

        public string Parish { get; }

        public int ArchdeaconryId { get; }

        public string Archdeaconry { get; }

        public Payment(
            int id,
            int amountPerYear,
            DateTime date,
            int? receiptNumber,
            int congregationId,
            string congregation,
            int parishId,
            string parish,
            int archdeaconryId,
            string archdeaconry)
        {
            Id = id;
            Amount = amountPerYear;
            Date = date;
            ReceiptNumber = receiptNumber;
            CongregationId = congregationId;
            Congregation = congregation;
            ParishId = parishId;
            Parish = parish;
            ArchdeaconryId = archdeaconryId;
            Archdeaconry = archdeaconry;
        }

        // for Dapper
        public Payment()
        {
        }
    }
}

File: SrdDatabase/ClientApp/src/models/payment.ts
import { PagedParameters, PagedResults } from "./shared";

export interface Payment {
    id?: number;
    amount?: number;
    congregationId?: number;
    congregation?: string;
    parishId?: number;
    parish?: string;
    archdeaconryId?: number;
    archdeaconry?: string;
    date?: Date;
    receiptNumber?: number;
    createdBy?: number;
}

export interface PaymentToSend extends Omit<Payment, "date"> {
    date?: string;
}

export interface PaymentDetails {
    payment: Payment;
}

export interface PaymentParameters extends PagedParameters {
    parishId?: number;
    archdeaconryId?: number;
    congregationId?: number;
    startDate?: Date;
    endDate?: Date;
    receiptNumber?: number;
};

export interface PaymentParametersToSend extends Omit<PaymentParameters, "startDate" | "endDate"> {
    startDate?: string;
    endDate?: string;
};

export interface PaymentResults extends PagedResults {
    payments: Payment[];
}
